diff -rupN bin/kafka-console-consumer.sh bin/kafka-console-consumer.sh
--- bin/kafka-console-consumer.sh	2014-02-20 06:46:38.635706202 +0000
+++ bin/kafka-console-consumer.sh	2014-02-20 06:41:29.362715093 +0000
@@ -14,5 +14,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-export KAFKA_HEAP_OPTS="-Xmx512M"
+if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
+    export KAFKA_HEAP_OPTS="-Xmx512M"
+fi
+
 exec $(dirname $0)/kafka-run-class.sh kafka.consumer.ConsoleConsumer $@
diff -rupN bin/kafka-console-producer.sh bin/kafka-console-producer.sh
--- bin/kafka-console-producer.sh	2014-02-20 06:46:38.635706202 +0000
+++ bin/kafka-console-producer.sh	2014-02-20 06:41:47.267004035 +0000
@@ -14,5 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-export KAFKA_HEAP_OPTS="-Xmx512M"
+if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
+    export KAFKA_HEAP_OPTS="-Xmx512M"
+fi
 exec $(dirname $0)/kafka-run-class.sh kafka.producer.ConsoleProducer $@
diff -rupN bin/kafka-consumer-perf-test.sh bin/kafka-consumer-perf-test.sh
--- bin/kafka-consumer-perf-test.sh	2014-02-20 06:46:38.635706202 +0000
+++ bin/kafka-consumer-perf-test.sh	2014-02-20 06:42:04.715285619 +0000
@@ -14,5 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-export KAFKA_HEAP_OPTS="-Xmx512M"
+if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
+    export KAFKA_HEAP_OPTS="-Xmx512M"
+fi
 exec $(dirname $0)/kafka-run-class.sh kafka.perf.ConsumerPerformance $@
diff -rupN bin/kafka-producer-perf-test.sh bin/kafka-producer-perf-test.sh
--- bin/kafka-producer-perf-test.sh	2014-02-20 06:46:38.635706202 +0000
+++ bin/kafka-producer-perf-test.sh	2014-02-20 06:42:43.499911532 +0000
@@ -14,5 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-export KAFKA_HEAP_OPTS="-Xmx512M"
+if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
+    export KAFKA_HEAP_OPTS="-Xmx512M"
+fi
 exec $(dirname $0)/kafka-run-class.sh kafka.perf.ProducerPerformance $@
diff -rupN bin/kafka-run-class.sh bin/kafka-run-class.sh
--- bin/kafka-run-class.sh	2014-02-20 06:46:38.635706202 +0000
+++ bin/kafka-run-class.sh	2014-02-20 06:43:41.572848724 +0000
@@ -22,12 +22,6 @@ fi
 
 base_dir=$(dirname $0)/..
 
-# create logs directory
-LOG_DIR=$base_dir/logs
-if [ ! -d $LOG_DIR ]; then
-	mkdir $LOG_DIR
-fi
-
 if [ -z "$SCALA_VERSION" ]; then
 	SCALA_VERSION=2.8.0
 fi
diff -rupN bin/kafka-server-start.sh bin/kafka-server-start.sh
--- bin/kafka-server-start.sh	2014-02-20 06:46:38.635706202 +0000
+++ bin/kafka-server-start.sh	2014-02-20 06:44:30.305635184 +0000
@@ -20,6 +20,13 @@ then
 	exit 1
 fi
 base_dir=$(dirname $0)
-export KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:$base_dir/../config/log4j.properties"
-export KAFKA_HEAP_OPTS="-Xmx1G -Xms1G"
+
+if [ "x$KAFKA_LOG4J_OPTS" = "x" ]; then
+    export KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:$base_dir/../config/log4j.properties"
+fi
+
+if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
+    export KAFKA_HEAP_OPTS="-Xmx1G -Xms1G"
+fi
+
 exec $base_dir/kafka-run-class.sh daemon kafkaServer kafka.Kafka $@
diff -rupN bin/kafka-simple-consumer-perf-test.sh bin/kafka-simple-consumer-perf-test.sh
--- bin/kafka-simple-consumer-perf-test.sh	2014-02-20 06:46:38.635706202 +0000
+++ bin/kafka-simple-consumer-perf-test.sh	2014-02-20 06:44:54.682028575 +0000
@@ -14,5 +14,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-export KAFKA_HEAP_OPTS="-Xmx512M"
+if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
+    export KAFKA_HEAP_OPTS="-Xmx1G -Xms1G"
+fi
+
 exec $(dirname $0)/kafka-run-class.sh kafka.perf.SimpleConsumerPerformance $@
diff -rupN bin/kafka-simple-consumer-shell.sh bin/kafka-simple-consumer-shell.sh
--- bin/kafka-simple-consumer-shell.sh	2014-02-20 06:46:38.635706202 +0000
+++ bin/kafka-simple-consumer-shell.sh	2014-02-20 06:45:04.914193704 +0000
@@ -14,4 +14,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
+    export KAFKA_HEAP_OPTS="-Xmx1G -Xms1G"
+fi
+
 exec $(dirname $0)/kafka-run-class.sh kafka.tools.SimpleConsumerShell $@
